File: python/unidad_03/problema_3_06.py
'''
El programa obtiene el mayor y menor de N números enteros que se ingresan.
'''

def obtener_mayor_menor(lista_datos):

    if lista_datos[0] > lista_datos[1]:

        numero_mayor = lista_datos[0]
        numero_menor = lista_datos[1]

    else:

        numero_mayor = lista_datos[1]
        numero_menor = lista_datos[0]

    for i in range(2, len(lista_datos)):

        if lista_datos[i] > numero_mayor:

            numero_mayor = lista_datos[i]

        elif lista_datos[i] < numero_menor:

            numero_menor = lista_datos[i]

    print(f"Número mayor: {numero_mayor}")

    print(f"Número menor: {numero_menor}")

def main():

    try:

        numero_datos = int(input("Número de datos: "))

        if numero_datos < 2:

            print("Error: el número de datos debe ser mayor que uno.")
            return

        lista_datos = []

        for i in range(numero_datos):

            numero = int(input(f"Dato {i + 1}: "))
            lista_datos.append(numero)

    except ValueError as e:

        print(e)
        return

    obtener_mayor_menor(lista_datos)

if __name__ == '__main__':

    main()

File: java/unidad_03/Problema_3_06/src/Problema_3_06.java
import java.util.InputMismatchException;
import java.util.Scanner;

/*
 * El programa obtiene el mayor y menor de N números enteros que se ingresan.
 */

public class Problema_3_06 {

    public static void obtenerMayorMenor(int[] arregloDatos) {

        int numeroMayor;
        int numeroMenor;

        if (arregloDatos[0] > arregloDatos[1]) {

            numeroMayor = arregloDatos[0];
            numeroMenor = arregloDatos[1];

        } else {

            numeroMayor = arregloDatos[1];
            numeroMenor = arregloDatos[0];
        }
        for (int i = 2; i < arregloDatos.length; i++) {

            if (arregloDatos[i] > numeroMayor) {

                numeroMayor = arregloDatos[i];

            } else if (arregloDatos[i] < numeroMenor) {

                numeroMenor = arregloDatos[i];
            }
        }
        System.out.printf("Número mayor: %d %n", numeroMayor);

        System.out.printf("Número menor: %d %n", numeroMenor);
    }

    public static void main(String[] args) {

        int numeroDatos;
        int arregloDatos[];

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Número de datos: ");
            numeroDatos = scanner.nextInt();

            if (numeroDatos < 2) {

                System.out.println("Error: el número de datos debe ser mayor que uno.");
                return;
            }
            arregloDatos = new int[numeroDatos];

            for (int i = 0; i < arregloDatos.length; i++) {

                System.out.printf("Dato %d: ", i + 1);
                arregloDatos[i] = scanner.nextInt();
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        obtenerMayorMenor(arregloDatos);
    }
}